sys/linux/init.go

```diff
@@ -299,6 +303,43 @@ func neutralizeSchedAttr(a prog.Arg) {
 	}
 }
 
+var dropMountErrorsRe = regexp.MustCompile(`(?m)(^\s*errors=[^,]+,?\s*|,?\s*errors=[^,]+)`)
+
+func neutralizeMount(c *prog.Call) {
+	typePtr, ok := c.Args[2].(*prog.PointerArg)
+	if !ok || typePtr.Res == nil {
+		return
+	}
+	typeData, ok := typePtr.Res.(*prog.DataArg)
+	if !ok {
+		return
+	}
+	// Only patch ext* mounts for now.
+	if !strings.HasPrefix(string(typeData.Data()), "ext") {
+		return
+	}
+	optsPtr, ok := c.Args[4].(*prog.PointerArg)
+	if !ok {
+		return
+	}
+	opts := ""
+	if optsPtr.Res != nil {
+		optsPtrData, ok := optsPtr.Res.(*prog.DataArg)
+		if !ok {
+			return
+		}
+		opts = string(optsPtrData.Data())
+	}
+	hadRemount := strings.Contains(opts, "errors=remount-ro")
+	opts = dropMountErrorsRe.ReplaceAllString(opts, "")
+	if hadRemount {
+		opts = "errors=remount-ro," + opts
+	} else {
+		opts = "errors=continue," + opts
```

Comment:
We do it here:
https://github.com/google/syzkaller/blob/master/executor/common_linux.h#L3044-L3056
because fuzzer generally can't guarantee contents of data in memory (can overlap with other arguments, or be overwritten by concurrent syscall args/results).

Reply:
Yes, we do sanitize `syz_mount_image`, but not `mount` directly, which we also have in the descriptions.